src/backend/catalog/storage_pending_deletes_redo.c

```diff
@@ -0,0 +1,324 @@
+/*-------------------------------------------------------------------------
+ *
+ * storage_pending_deletes_redo.c
+ *	  code to support processing of pending deletes (orphaned files) in WAL
+ *
+ * Copyright (c) 2025 Greengage Community
+ *
+ *	  src/backend/catalog/storage_pending_deletes_redo.c
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/clog.h"
+#include "access/transam.h"
+#include "catalog/storage_pending_deletes_redo.h"
+#include "miscadmin.h"
+#include "storage/md.h"
+#include "utils/elog.h"
+#include "utils/guc.h"
+#include "utils/hsearch.h"
+
+/*
+ * HTAB entry for pending deletes for the given xid.
+ */
+typedef struct PendingDeleteHtabNode
+{
+	TransactionId xid;
+	List	   *relnode_list;	/* list of RelFileNodePendingDelete */
+}	PendingDeleteHtabNode;
+
+/*
+ * Hash table for pending deletes relfilenodes for a given xid.
+ */
+static HTAB *pendingDeletesRedo = NULL;
+
+/*
+ * This function inserts XLOG_PENDING_DELETE record into WAL.
+ */
+void
+PdlXLogInsert()
+{
+	Size		size = 0;
+
+	if (IsBootstrapProcessingMode() || !gp_track_pending_delete)
+		return;
+
+	PendingRelXactDeleteArray *arr = PdlXLogShmemDump(&size);
+
+	if (size > 0 && arr != NULL)
+	{
+		XLogRecPtr	rec;
+		XLogRecData rdata =
+		{
+			.buffer = InvalidBuffer,
+			.data = (char *) arr,
+			.len = size,
+			.next = NULL,
+			.buffer_std = false
+		};
+
+		rec = XLogInsert(RM_XLOG_ID, XLOG_PENDING_DELETE, &rdata);
+
+		XLogFlush(rec);
```

Comment:
The rec variable is used once only. Let's remove it.
```suggestion
		XLogRecData rdata =
		{
			.buffer = InvalidBuffer,
			.data = (char *) arr,
			.len = size,
			.next = NULL,
			.buffer_std = false
		};

		XLogFlush(XLogInsert(RM_XLOG_ID, XLOG_PENDING_DELETE, &rdata));
```

Reply:
The proposed change doesn't introduce any memory/time optimization. But it reduces code readability. And also introduces difficulties during debug if one wants to set a breakpoint on insert before flush. I prefer to leave it as is.